src/SonarScanner.MSBuild.Common/Telemetry/TelemetryUtils.cs

```diff
@@ -22,17 +22,78 @@ namespace SonarScanner.MSBuild.Common;
 
 public static class TelemetryUtils
 {
-    private static string[] telemetryProperties = [SonarProperties.ScanAllAnalysis];
-
     public static void AddTelemetry(ILogger logger, AggregatePropertiesProvider aggregatedProperties)
     {
-        foreach (var propertyWithProvider in aggregatedProperties.GetAllPropertiesWithProvider().Where(x => !x.Key.ContainsSensitiveData() && telemetryProperties.Contains(x.Key.Id)))
+        foreach (var kvp in aggregatedProperties.GetAllPropertiesWithProvider().SelectMany(SelectMany))
+        {
+            logger.AddTelemetryMessage(kvp.Key, kvp.Value);
+        }
+    }
+
+    private static IEnumerable<KeyValuePair<string, string>> SelectMany(KeyValuePair<Property, IAnalysisPropertyProvider> argument)
```

Comment:
```suggestion
    private static IEnumerable<KeyValuePair<string, string>> FilterTelemetryProperties(KeyValuePair<Property, IAnalysisPropertyProvider> argument)
```
or something similar?